### ISSUE TITLE
fqcn[action-core] doesn't correctly detect ansible.builtin.yum, insists I use ansible.builtin.dnf even when my target server doesn't have dnf installed

### ISSUE THREAD

**Issue by Endemoniada** [ansible/ansible-lint#4251]
##### Summary

When linting code that uses the module `ansible.builtin.yum` using the full FQCN, ansible-lint will insist that it isn't using the FQCN and that I use `ansible.builtin.dnf` instead, even though I am writing tasks that target RHEL7 where dnf isn't even installed.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 24.6.1 using ansible-core:2.17.1 ansible-compat:24.6.1 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

Running in WSL on ubi9:

```console
$ cat /etc/*release
NAME="Red Hat Enterprise Linux"
VERSION="9.4 (Plow)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="9.4"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Red Hat Enterprise Linux 9.4 (Plow)"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:redhat:enterprise_linux:9::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_BUGZILLA_PRODUCT_VERSION=9.4
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.4"
Red Hat Enterprise Linux release 9.4 (Plow)
Red Hat Enterprise Linux release 9.4 (Plow)
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```yaml (paste below)
---
# tasks file for install

- name: "Install RPM package"
  ansible.builtin.yum:
    name: "{{ kibana_install_rpm_url }}"
  become: true
```

##### Desired Behavior

ansible-lint should have no issue with this code, since it actually does use FQCN. It should also respect that yum is still in use, even on systems that do have dnf installed, and especially on legacy servers that do not support dnf.

##### Actual Behavior

```console
WARNING  Listing 1 violation(s) that are fatal
fqcn[action-core]: Use FQCN for builtin module actions (ansible.builtin.yum).
tasks/main.yml:4 Use `ansible.builtin.dnf` or `ansible.legacy.dnf` instead.
```

Downgrading to ansible-lint 6.22.2 (by way of running Python 3.9 and ansible 2.15 when installing ansible-lint) fixes this issue, it no longer incorrectly complains about missing FQCN.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Qalthos** [comment]
As of ansible 2.17, the `yum` module no longer exists, as a part of dropping Python 2 support for remote hosts (see [2.17 changelog](https://github.com/ansible/ansible/blob/v2.17.0/changelogs/CHANGELOG-v2.17.rst#removed-features-previously-deprecated), https://github.com/ansible/ansible/pull/81895). This does not help you, as the `dnf` module is, as you note, incompatible with old `yum`.

Unfortunately, as a consequence of all of this, the only way to prevent the warning is to keep ansible-lint's version of ansible below 2.17. There is nothing we can do about the message because we are limited to the information we get from ansible itself, and `yum` is now just an alias for `dnf`.
